src/yachtzee/fishy.ts

```diff
@@ -0,0 +1,203 @@
+      },
+    },
+    {
+      name: "powdered candy sushi set",
+      cost: mallPrice($item`powdered candy sushi set`) + bestWaterBreathingEquipment.cost,
+      action: () => {
+        acquire(
+          1,
+          $item`powdered candy sushi set`,
+          1.2 * mallPrice($item`powdered candy sushi set`)
+        );
+        if (!have($item`powdered candy sushi set`)) {
+          throw new Error("Unable to obtain powdered candy sushi set");
+        }
+        use(1, $item`powdered candy sushi set`);
+      },
+    },
+    {
+      name: "concentrated fish broth",
+      cost: mallPrice($item`concentrated fish broth`) + bestWaterBreathingEquipment.cost,
+      action: () => {
+        acquire(1, $item`concentrated fish broth`, 1.2 * mallPrice($item`concentrated fish broth`));
+        if (!have($item`concentrated fish broth`)) {
+          throw new Error("Unable to obtain concentrated fish broth");
+        }
+        use(1, $item`concentrated fish broth`);
+      },
+    },
+    {
+      name: "Lutz, the Ice Skate",
+      cost:
+        get("_skateBuff1", false) || get("skateParkStatus") !== "ice"
+          ? Infinity
+          : bestWaterBreathingEquipment.cost,
+      action: () => {
+        cliExecute("skate lutz");
+      },
+    },
+    {
+      name: "The Haggling",
+      cost: canAdventure($location`The Brinier Deepers`)
+        ? (have($effect`Lucky!`) ? 0 : mallPrice($item`11-leaf clover`)) +
+          get("valueOfAdventure") +
+          bestWaterBreathingEquipment.cost +
+          fishyCloverAdventureOpportunityCost(haveFishyPipe)
+        : Infinity,
+      action: () => {
+        if (!have($effect`Lucky!`)) {
+          acquire(1, $item`11-leaf clover`, 1.2 * mallPrice($item`11-leaf clover`));
+          if (!have($item`11-leaf clover`)) {
+            throw new Error("Unable to get 11-leaf clover for fishy!");
+          }
+          use(1, $item`11-leaf clover`);
+        }
+        if (haveFishyPipe) use(1, $item`fishy pipe`);
+        adventureMacro($location`The Brinier Deepers`, Macro.abort());
+        if (get("lastAdventure") !== "The Brinier Deepers") {
+          print(
+            "We failed to adventure in The Brinier Deepers, even though we thought we could. Try manually adventuring there for a lucky adventure.",
+            "red"
+          );
+        }
+        if (haveEffect($effect`Fishy`) < yachtzeeTurns) {
+          throw new Error("Failed to get fishy from clover adv");
+        }
+      },
+    },
+    {
+      name: "Just Fishy Pipe",
+      cost:
+        (haveFishyPipe ? 10 : haveEffect($effect`Fishy`)) + (pyecAvailable() ? 5 : 0) <
+        yachtzeeTurns
+          ? Infinity
+          : 0,
+      action: () => {
+        if (haveFishyPipe && haveEffect($effect`Fishy`) < yachtzeeTurns) use(1, $item`fishy pipe`);
+      },
+    },
+  ];
+
+  const bestFishySource = fishySources.reduce((left, right) => {
+    return left.cost < right.cost ? left : right;
+  });
```

Comment:
Correspondingly, if you're refactoring the above to add the `fishyTurns` field, we would conduct the filtering here to remove sources that don't provide sufficient fishy turns to cover our yachtzee turns